src/core/execute.c

```diff
@@ -4336,6 +4344,14 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                 fprintf(f, "%sCPUAffinity: %s\n", prefix, affinity);
         }
 
+        if (numa_policy_type_valid(c->numa_policy.type)) {
+                _cleanup_free_ char *nodes = NULL;
+
+                nodes = cpu_set_to_range_string(&c->numa_policy.nodes);
+                fprintf(f, "%sNUMAPolicy: %s\n", prefix, numa_policy_type_to_string(c->numa_policy.type));
+                fprintf(f, "%sNUMAMask: %s\n", prefix, nodes);
```

Comment:
use strnull(nodes) or so, in case the cpu_set_to_range_string() memory allocation failed). in some cases printf/fprintf doesn't like NULL at all for %s, hence let's avoid it